ext/drmgr/drmgr.c

```diff
@@ -2506,6 +2524,78 @@ drmgr_pop_cls(void *drcontext)
     return drmgr_cls_stack_pop();
 }
 
+/***************************************************************************
+ * Low-on-memory
+ */
+
+DR_EXPORT
+bool
+drmgr_register_low_on_memory_event(void (*func)())
+{
+    return drmgr_generic_event_add(&cblist_low_on_memory, low_on_memory_event_lock,
+                                   (void (*)(void))func, NULL, false, NULL);
+}
+
+DR_EXPORT
+bool
+drmgr_register_low_on_memory_event_ex(void (*func)(), drmgr_priority_t *priority)
```

Comment:
Could we simplify for all new registration routines and have the main one take the priority and eliminate a separate _ex version?  Normally an _ex version is added only if we've already had a base version in the API and we can't change it for backward compatibility.  For that matter maybe we could have just one version with the user data too.   The downside is the names are a little different when passing user data vs the older ones, but IMHO that's balanced out by having just one function instead of 3 (and just one unreg instead of 2).  WDYT?

Reply:
When you say "main one" are you referring to the following API function?

```
bool
 drmgr_register_low_on_memory_event_user_data(void (*func)(void *user_data),
                                              drmgr_priority_t *priority, void *user_data);
```

Reply:
i.e. do you just want the above API function? Otherwise, you cannot just have one unregister function because callbacks still have diff prototypes.

Reply:
Yes, just the callback type with all of the options: priority and user_data.

Reply:
By "main one" I mean the one with the plain name.  I.e., no _ex suffix.

Reply:
I understand that the excessive API functions are a problem, but tbh, thinking about it now, I am not sure if we should be changing the structure of API prototypes in relation to older ones. I think the user benefits with similar API structure.  Also, the extra functions don't really consist of a lot of code (most are one liners) so I don't think it is really a problem for maintainability in the current state. 

Having said that, I am not hell-bent on the issue you; just throwing it out there. LMK

Reply:
Maybe get some more opinions since it's a user-facing thing.  @hgreving2304, @Carrotman42 -- any opinions one way or the other?